internal/pkg/service/stream/dependencies/api.go

```diff
@@ -69,6 +69,5 @@ func (v *apiScope) APIPublicURL() *url.URL {
 }
 
 func (v *apiScope) HTTPSourcePublicURL() *url.URL {
-	out, _ := url.Parse(v.httpSourcePublicURL.String()) // clone
-	return out
+	return v.httpSourcePublicURL
```

Comment:
because the parser is in the function:
```
func (s *Source) FormatHTTPSourceURL(httpSourcePublicURL string) (string, error) {
	u, err := url.Parse(httpSourcePublicURL)
	if err != nil {
		return "", err
	}

	return u.JoinPath("stream", s.ProjectID.String(), s.SourceID.String(), s.HTTP.Secret).
		String(), nil
}
```